service/history/historyEngine.go

```diff
@@ -1012,6 +1010,39 @@ func (e *historyEngineImpl) RecordActivityTaskStarted(ctx context.Context,
 	return response, err
 }
 
+type decisionBlobSizeChecker struct {
+	sizeLimitWarn  int
+	sizeLimitError int
+	domainID       string
+	workflowID     string
+	runID          string
+	metricsClient  metrics.Client
+	logger         bark.Logger
+	msBuilder      mutableState
+	completedID    int64
+}
+
+func (c *decisionBlobSizeChecker) failWorkflowIfBlobSizeExceedsLimit(blob []byte) (bool, error) {
```

Comment:
Can you also pass in a reason along with blob, so instead of having a generic 'FailureReasonDecisionBlobSizeExceedsLimit' we can have something finer grained like you had in your previous diff.